### PR TITLE
Compatibility with Tiny Tiny RSS v21.03-2f402d598

### PR DESCRIPTION
**Pull request by qiwichupa**
Restored compatibility with current tt-rss.
- new function csrf_ignore (without this plugin isn't working)
- fix in "Ajax" part (some new way to submit settings)
- quotes for php constants (php requirements)

### CODE REVIEW

File: init.php [hunk 1]
```diff
@@ -1,15 +1,16 @@
 <?php
+# https://github.com/klempin/af_refspoof
```
Comment (klempin) [inline init.php:2]: This comment is not needed.

File: init.php [hunk 2]
```diff
@@ -26,7 +27,7 @@ public function init($host)
 
     public function hook_prefs_edit_feed($feedId)
     {
-        $enabledFeeds = $this->host->get($this, STORAGE_ENABLED_FEEDS, array());
+        $enabledFeeds = $this->host->get($this, "STORAGE_ENABLED_FEEDS", array());
```
Comment (klempin) [inline init.php:30]: This changes the type of the argument from a constant to a string. To fix the issue this should be replaced with `static::STORAGE_ENABLED_FEEDS` instead. This applies to all occurences of `STORAGE_ENABLED_FEEDS` and `STORAGE_ENABLED_DOMAINS`

File: init.php [hunk 3]
```diff
@@ -47,51 +48,45 @@ public function hook_prefs_edit_feed($feedId)
...
-        }
+        if ($args != "prefFeeds") return;
 
         $title = __("Fake referral");
         $header = __("Enable referral spoofing based on the feed domain (enter one domain per line)");
-        $enabledDomains = implode("\n", $this->host->get($this, STORAGE_ENABLED_DOMAINS, ""));
+        $enabledDomains = implode("\n", $this->host->get($this, "STORAGE_ENABLED_DOMAINS", ""));
         $button = __("Save");
-
+        $value = $this->host->get($this, "af_refspoof");
```
Comment (klempin) [inline init.php:57]: Is this variable needed?

File: init.php [hunk 4]
```diff
@@ -47,51 +48,45 @@ public function hook_prefs_edit_feed($feedId)
...
-        <input type="hidden" data-dojo-type="dijit/form/TextBox" name="plugin" value="af_refspoof">
-        <input type="hidden" data-dojo-type="dijit/form/TextBox" name="method" value="saveDomains">
+
+    <form dojoType="dijit.form.Form" style="flex-grow:1;display:flex;flex-direction:column;">
+        <input type="hidden" dojoType="dijit.form.TextBox" name="op" value="pluginhandler">
+        <input type="hidden" dojoType="dijit.form.TextBox" name="plugin" value="af_refspoof">
+        <input type="hidden" dojoType="dijit.form.TextBox" name="method" value="saveDomains">
         <script type="dojo/method" event="onSubmit" args="evt">
             evt.preventDefault();
+            console.log(dojo.objectToQuery(this.getValues()));
```
Comment (klempin) [inline init.php:69]: This should be removed.

File: init.php [hunk 5]
```diff
@@ -178,12 +175,17 @@ public function api_version()
...
 
         if (strpos($host, "www.") === 0 && in_array(str_replace("www.", "", $host), $enabledDomains)) {
             return true;
         }
         return in_array($host, $enabledDomains);
     }
+
+    public function csrf_ignore($method) {
+        return true;
+    }
```
Comment (klempin) [inline init.php:189]: It is safer to only ignore CSRF on specific requests. This function should only return true if `$method` is `proxy`. All other methods should check CSRF tokens normally.

**Review by klempin — CHANGES_REQUESTED** [review]
Hi, thanks for the pull request! This PR fixes quite a few issues that have arisen due to code changes in TT-RSS and newer PHP versions. I have a few changes I would like to request before merging. There is also one minor merge conflict due to a change I made, today.

File: init.php (hunk 3 above)
Comment (qiwichupa) [reply init.php:57]: No

**Pull request closed without merging**